src/components/common/Typography/Typography.tsx

```diff
@@ -0,0 +1,110 @@
+import styled from '@emotion/styled';
+import type { ReactNode, ElementType } from 'react';
+import type { Theme } from '@emotion/react';
+import { theme } from '@styles/tokens';
+
+type Props = {
+  children: ReactNode;
+  variant?: VariantKeys;
+  color?: ColorKeys;
+  as?: HTMLTags;
+  width?: string;
+  textAlign?: 'left' | 'center' | 'right';
+} & React.HTMLAttributes<HTMLElement>;
+
+export const Typography = ({
+  children,
+  variant = 'body1Regular',
+  color = 'textDefault',
+  as = 'p',
+  width,
+  textAlign = 'left',
+  ...rest
+}: Props) => {
+  return (
+    <StyledTypography
+      as={as as ElementType}
+      variant={variant}
+      colorKey={color}
+      width={width}
+      textAlign={textAlign}
+      {...rest}
+    >
+      {children}
+    </StyledTypography>
+  );
+};
+
+const StyledTypography = styled.p<{
+  variant: VariantKeys;
+  colorKey: ColorKeys;
+  width?: string;
+  textAlign: 'left' | 'center' | 'right';
+}>(({ theme, variant, colorKey, width, textAlign }) => ({
+  ...theme.typography[variant],
+  color: getColor(colorKey, theme),
+  margin: 0,
+  width,
+  textAlign,
+}));
+
+// theme.colors 구조에 기반한 ColorKey 분해
+type GrayColorKeys = keyof typeof theme.colors.gray;
+type YellowColorKeys = keyof typeof theme.colors.yellow;
+type BrownColorKeys = keyof typeof theme.colors.brown;
+type BlueColorKeys = keyof typeof theme.colors.blue;
+type RedColorKeys = keyof typeof theme.colors.red;
+type SemanticKeys = keyof typeof theme.colors.semantic;
+
+type ColorKeys =
+  | GrayColorKeys
+  | YellowColorKeys
+  | BrownColorKeys
+  | BlueColorKeys
+  | RedColorKeys
+  | SemanticKeys;
+
+type VariantKeys = keyof typeof theme.typography;
+type HTMLTags =
+  | 'h1'
+  | 'h2'
+  | 'h3'
+  | 'h4'
+  | 'h5'
+  | 'h6'
+  | 'p'
+  | 'span'
+  | 'div'
+  | 'label';
```

Comment:
`React.ElementType` 타입을 쓰시면 as prop의 타입을 쉽게 지정할 수 있어요.
[Extract](https://www.typescriptlang.org/docs/handbook/utility-types.html#extracttype-union) 유틸리티 타입을 사용해 `ElementType` 중 `Typography` 컴포넌트가 받아들일 수 있는 요소를 지정하면 더 좋을 것 같습니다

```suggestion
type HTMLTags = Extract<ElementType, ...>;
```